use-cases.md

```diff
@@ -119,9 +126,28 @@ After a customer has checked out from a reservation, reputation management syste
 
-## Mobile Key Systems
+### Marketing Opt-out
+
+A customer has the ability to opt-out of marketing communciation. As Reputation management systems are also commonly used to launch marketing campaigns, the integration should ensure that if the option, `SendMarketingEmails` in the customer object is non existant then the customer is unsubscribed to any marketing.
+
+## Mobile key systems
 
 Mobile Key solutions require a state of reservation in real time, a key should not be issued to a guest until they have been checked in Mews. To avoid polling for updated reservations, a Reservation Websocket should be used.
 
 After receiving a websocket event, use [Get all reservations by ids](operations/reservations.md#get-all-reservations-by-ids) to retrieve information about the reservation and customer if the websocket event fits your criteria. With this response, you will have information to issue the mobile key to the customer using their contact information in Mews.
 
+## Customer relationship management systems
+
+### Inital Data Pull
+
+Performed once when setting up the connection, because the CRM needs to obtain existing customers and previous reservtions. The CRM should obtain the customers and reservations in time-limited batches using [Get all reservations](operations/reservations.md#get-all-reservations) and [Get all customers](operations/reservations.md#get-all-customers) with the time filter set to `created` \(that will give you all customers and reservations which were created in the selected interval\). Size of the batches depends on size of the hotel and its occupancy, but in general **weekly batches** are recommended and should work well even for big hotels \(1000+ units\). In order to get all customers and reservations e.g. in the past year, the CRM should call [Get all customers](operations/reservations.md#get-all-customers) and [Get all reservations](operations/reservations.md#get-all-reservations) sequentially 52 times (one call for each week in the past year). That would give the CRM all customers and all reservations that have been created within the past year. To obtain products associated with reservations e.g. a breakfast, `Items` should be set to `true` in the `Extent` parameter.
+
+One can take advantage of the fact that reservations are usually booked a few weeks or months in advance. The further in future, the lower the occupancy, so the reservation batch length may increase with the distance to future from current date. E.g. weekly batches can be used only for the first three months of the future year when there is higher occupancy. And for the remaining 9 months, monthly batches would be sufficient. This would reduce the operation count from 52 to 21 \(12 weekly batches + 9 monthly batches\).
```

Comment:
One can take advantage of the fact that reservations are usually booked a few weeks or months in advance. The further in the future, the lower the occupancy, so the reservation batch length may increase with the distance into future from current date. E.g. weekly batches can be used only for the first three months of the future year when there is higher occupancy. And for the remaining 9 months, monthly batches would be sufficient. This would reduce the operation count from 52 to 21 \(12 weekly batches + 9 monthly batches\).